### DISCUSSION TITLE
Which type of learning rule is used in eth_mnist example?

### DISCUSSION THREAD

**Discussion by rafaelblevin821** [BindsNET/bindsnet#556] [General]
I understand it's based on the Diehl and Cook model, but I cannot see anywhere in the code where it specifies either STDP or PosPre? Does the learning rule automatically come built into the Diehl and Cook imported model?

**Comment by SimonInParis** [comment]
Yes, the Diehl and Cook model is defined in models.py, and the learning
rule rule is PostPre, just as you thought:
# Connections
w = 0.3 * torch.rand(self.n_inpt, self.n_neurons)
input_exc_conn = Connection(
source=input_layer,
target=exc_layer,
w=w,
update_rule=PostPre,
nu=nu,
reduction=reduction,
wmin=wmin,
wmax=wmax,
norm=norm,
)

**Reply by rafaelblevin821** [reply]
Hi, thanks for clarifying. I still cannot find this in the documentation though.

**Reply by Hananel-Hazan** [reply]
Thanks @SimonInParis 
Here is the reference @rafaelblevin821 
[documentation ](https://github.com/BindsNET/bindsnet/blob/bc26497b64a09e9166af9c8b2e81f80b73806d15/bindsnet/models/models.py#L87) and the lines Simon [pointing out](https://github.com/BindsNET/bindsnet/blob/bc26497b64a09e9166af9c8b2e81f80b73806d15/bindsnet/models/models.py#L169)

**Comment by rafaelblevin821** [comment]
Thank you both for the information.
